packages/validator/src/services/attestationDuties.ts

```diff
@@ -63,12 +63,32 @@ export class AttestationDutiesService {
 
     if (metrics) {
       metrics.attesterDutiesCount.addCollect(() => {
+        const currentSlot = this.clock.getCurrentSlot();
         let duties = 0;
-        for (const attDutiesAtEpoch of this.dutiesByIndexByEpoch.values()) {
+        let nextDutySlot = null;
+        for (const [epoch, attDutiesAtEpoch] of this.dutiesByIndexByEpoch) {
           duties += attDutiesAtEpoch.dutiesByIndex.size;
+
+          // Historical epochs can be skipped when determining next duty slot
+          if (epoch < this.clock.currentEpoch) continue;
+
+          // There is no need to check every single duty at a certain amount
```

Comment:
I'm not sure the logic here, if `dutiesByIndex.size >= 64` it's not a guarantee we'll have duties in the next slot

Reply:
Yes it is not a guarantee even it this number was a lot higher but I found that at a certain amount of validators the metrics becomes pretty useless.

The main goal of this check is to prevent going through all duties if validator count is really high. The operation in each loop is pretty cheap though. Let me know if you think this is not a performance problem.

Reply:
yes I think it's not a performance issue here, removing it looks good to me, also we skip epochs ahead if we found one so looks great overall